File: 350-intersection-of-two-arrays-ii/350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        # items that appear in both lists
        intersect = []
        i = 0
        j = 0
        while i < len(nums1):
            while j < len(nums2):
                if i >= len(nums1) or j >= len(nums2):
                    return intersect
                if nums1[i] == nums2[j]:
                    intersect.append(nums1[i])
                    nums1.pop(i)
                    nums2.pop(j)
                    j = 0
                    continue
                j += 1
            i += 1
            j = 0
        return intersect
                    

File: 350-intersection-of-two-arrays-ii/350-intersection-of-two-arrays-ii.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    let counter = {};
    let result = [];
    for (let num of nums1){
        if (counter[num]){
            counter[num] += 1;
        } else {
            counter[num] = 1;
        }
    }
    for (const num of nums2){
        if (counter[num] > 0){
            counter[num] -= 1;
            result.push(num);
        }
    }
    return result;
}